src/data.cc

```diff
@@ -145,7 +145,7 @@ data_reassign_sensors (data_t *self, bool is_propagation_needed)
         // discover the sub-type of the asset
         const char *subtype = fty_proto_aux_string (one_sensor, "subtype", "");
         // check if it is sensor or not
-        if ( !streq (subtype, "sensor") ) {
+        if ( !streq (subtype, "sensor") || !streq (subtype, "sensorgpio")) {
```

Comment:
Shouldn't this be `&&` ? If not this type of sensor AND not that type of sensor, then it is not any sensor...